crates/turbo-tasks/src/primitives.rs

```diff
@@ -6,37 +6,39 @@ use futures::TryFutureExt;
 use turbo_tasks_macros::primitive as __turbo_tasks_internal_primitive;
 
 use crate::{
-    TryJoinIterExt, Vc, {self as turbo_tasks},
+    RcStr, TryJoinIterExt, Vc, {self as turbo_tasks},
 };
 
 __turbo_tasks_internal_primitive!(());
 __turbo_tasks_internal_primitive!(String);
+__turbo_tasks_internal_primitive!(RcStr);
 
 #[turbo_tasks::function]
-fn empty_string() -> Vc<String> {
-    Vc::cell(String::new())
+fn empty_string() -> Vc<RcStr> {
+    Vc::cell(RcStr::from(""))
 }
 
-impl Vc<String> {
+impl Vc<RcStr> {
     #[deprecated(note = "use Default::default() instead")]
     #[inline(always)]
-    pub fn empty() -> Vc<String> {
+    pub fn empty() -> Vc<RcStr> {
         empty_string()
     }
 }
 
 __turbo_tasks_internal_primitive!(Option<String>);
-__turbo_tasks_internal_primitive!(Vec<String>);
+__turbo_tasks_internal_primitive!(Option<RcStr>);
+__turbo_tasks_internal_primitive!(Vec<RcStr>);
 
 #[turbo_tasks::function]
-fn empty_string_vec() -> Vc<Vec<String>> {
+fn empty_string_vec() -> Vc<Vec<RcStr>> {
     Vc::cell(Vec::new())
 }
 
 impl Vc<Vec<String>> {
     #[deprecated(note = "use Default::default() instead")]
     #[inline(always)]
-    pub fn empty() -> Vc<Vec<String>> {
+    pub fn empty() -> Vc<Vec<RcStr>> {
```

Comment:
Given that this is on the specific `Vc<Vec<String>>` type, I think this should continue returning `Self`.

If you want, you could change this to be an impl on `Vc<Vec<RcStr>>`.

If there's no remaining callsites (or just 1 or 2), we should just delete it.